elyra/tests/pipeline/kfp/conftest.py

```diff
@@ -0,0 +1,432 @@
+    }
+    runtime_config = create_runtime_config(
+        metadata_managers[Runtimes.RUNTIMES_SCHEMASPACE_NAME], runtime_config_options
+    )
+
+    # Create runtime image configurations
+    runtime_image_options = {"require_pull_secret": request.param.get("require_pull_secret")}
+    runtime_image_configs = create_runtime_image_configs(
+        metadata_managers[RuntimeImages.RUNTIMES_IMAGES_SCHEMASPACE_NAME],
+        pipeline_file=request.param["pipeline_file"],
+        pipeline_object=pipeline_object,
+        customization_options=runtime_image_options,
+    )
+
+    yield {
+        "pipeline_file": request.param["pipeline_file"],
+        "pipeline_object": pipeline_object,
+        "runtime_config": runtime_config,
+        "runtime_image_configs": runtime_image_configs,
+        "metadata_managers": metadata_managers,
+        "fixture_parameters": request.param,
+    }
+
+
+# ------------------------------
+# Supporting functions
+# ------------------------------
+
+
+def create_runtime_config(rt_metadata_manager: MetadataManager, customization_options: Dict[str, Any] = {}) -> Metadata:
+    """
+    Create a persisted Kubeflow Pipelines runtime configuration. The generated
+    runtime configuration is customizable using these options properties:
+        - "config_name": "custom-config-name",  # default is "test-config"
+        - "workflow_engine": WorkflowEngineType.XXX,  # Enum value, ARGO is default
+        - "use_cos_credentials_secret": True,  # or False (default)
+    """
+    kfp_runtime_config = {
+        "display_name": "Test runtime configuration",
+        "schema_name": "kfp",
+        "metadata": {
+            "display_name": "Test runtime configuration",
+            "runtime_type": "KUBEFLOW_PIPELINES",
+            "api_endpoint": "http://examples.com:31737",
+            "user_namespace": "kubeflow-user-example-com",
+            "auth_type": "DEX_STATIC_PASSWORDS",
+            "api_username": "user@example.com",
+            "api_password": "12341234",
+            "cos_endpoint": "http://examples.com:31671",
+            "cos_bucket": "test",
+            "tags": ["test"],
+        },
+    }
+
+    if customization_options.get("workflow_engine") == WorkflowEngineType.TEKTON:
+        kfp_runtime_config["metadata"]["engine"] = "Tekton"
+    else:
+        kfp_runtime_config["metadata"]["engine"] = "Argo"
+
+    if customization_options.get("use_cos_credentials_secret"):
+        kfp_runtime_config["metadata"]["cos_auth_type"] = "KUBERNETES_SECRET"
+        kfp_runtime_config["metadata"]["cos_username"] = "my_name"
+        kfp_runtime_config["metadata"]["cos_password"] = "my_password"
+        kfp_runtime_config["metadata"]["cos_secret"] = "secret-name"
+    else:
+        kfp_runtime_config["metadata"]["cos_auth_type"] = "USER_CREDENTIALS"
+        kfp_runtime_config["metadata"]["cos_username"] = "my_name"
+        kfp_runtime_config["metadata"]["cos_password"] = "my_password"
```

Comment:
I find this more clear but it might just be a personal preference thing 🙂 

```suggestion
    kfp_runtime_config["metadata"]["cos_username"] = "my_name"
    kfp_runtime_config["metadata"]["cos_password"] = "my_password"
    if customization_options.get("use_cos_credentials_secret"):
        kfp_runtime_config["metadata"]["cos_auth_type"] = "KUBERNETES_SECRET"
        kfp_runtime_config["metadata"]["cos_secret"] = "secret-name"
    else:
        kfp_runtime_config["metadata"]["cos_auth_type"] = "USER_CREDENTIALS"
```